rasa/core/training/story_reader/yaml_story_reader.py

```diff
@@ -0,0 +1,126 @@
+                else:
+                    logger.warning(f"Unexpected key {key} found in {self.source_name}")
+
+        except ValueError:
+            logger.error(f"Failed to read {filename}, it will be skipped")
+
+        return []
+
+    def _parse_stories(self, stories: List[Dict[Text, Any]]) -> None:
+
+        for story_item in stories:
+            if not isinstance(story_item, dict):
+                logger.warning(
+                    f"Unexpected block found in {self.source_name}: \n"
+                    f"{story_item}\n"
+                    f"Items under the `{KEY_STORIES}` key must be YAML dictionaries."
+                    f"It will be skipped."
+                )
+                continue
+
+            if KEY_PLAIN_STORY in story_item.keys():
+                self._parse_plain_story(story_item)
+
+    def _parse_plain_story(self, story_item: Dict[Text, Any]) -> None:
+        story_name = story_item.get(KEY_PLAIN_STORY, "")
+        steps = story_item.get(KEY_PLAIN_STORY_STEPS, "")
+
+        self._new_story_part(story_name, self.source_name)
+
+        for step in steps:
+            self._parse_step(step)
+
+    def _parse_step(self, step: Dict[Text, Any]):
+
+        if KEY_PLAIN_STORY_USER_UTTERANCE in step.keys():
+            self._parse_plain_story_user_utterance(step)
+        if KEY_SLOT_NAME in step.keys():
+            self._parse_slot(step)
+        if KEY_ACTION in step.keys():
+            self._parse_action(step)
+
+    def _parse_plain_story_user_utterance(self, step: Dict[Text, Any]):
+
+        user_utterance = step.get(KEY_PLAIN_STORY_USER_UTTERANCE, "").strip()
+        entities = step.get(KEY_PLAIN_STORY_ENTITIES, "")
+
+        if not user_utterance.startswith(INTENT_MESSAGE_PREFIX):
+            logger.warning(
+                f'User intent "{user_utterance}" should start with '
+                f'"{INTENT_MESSAGE_PREFIX}"'
+            )
+        else:
+            user_utterance = user_utterance[1:]
+
+        entities_list = []
+        for entity in entities:
+            if isinstance(entity, dict):
+                for key, value in entity.items():
+                    entities_list.append({"entity": key, "value": value})
+            else:
+                entities_list.append({"entity": entity, "value": ""})
+
+        intent = {"name": user_utterance, "confidence": 1.0}
+        utterance = UserUttered(user_utterance, intent, entities_list)
+
+        if self.use_e2e:
+            # TODO
+            pass
+        else:
+            self.current_step_builder.add_user_messages([utterance])
+
+    def _parse_slot(self, step: Dict[Text, Any]) -> None:
+
+        slot_name = step.get(KEY_SLOT_NAME, "")
+        slot_value = step.get(KEY_SLOT_VALUE, "")
+
+        self._add_event(SlotSet.type_name, {slot_name: slot_value})
+
+    def _parse_action(self, step: Dict[Text, Any]) -> None:
+
+        action_name = step.get(KEY_ACTION, "")
+        self._add_event(action_name, {})
```

Comment:
optiona: do we actually need the `StoryStepBuilder` actually? Codewise I think it would be nicer / simpler to not have the shared state as object attribute? Just some thoughts - could be an idea for a later refactoring.

Reply:
It has some useful functionality, e.g. it deals with the `OR` statements internally.